orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java

```diff
@@ -0,0 +1,559 @@
+                        "Can not insert or update association", sqle);
+            } */
+		 
+
+	}
+
+	@Override
+	public void removeAssociation(AssociationKey key, AssociationContext associationContext) {
+		log.info( "removeAssociation: AssociationKey:" + key + "; AssociationContext:" + associationContext + ";" );
+		/*
+		  Tuple outEntityTuple = associationContext.getEntityTuple(); String inClassName = key.getTable(); Object
+		  inBusinessPrimaryKeyName = EntityKeyUtil.findPrimaryKeyName( key.getEntityKey() ); Object
+		  inBusinessPrimaryKeyValue = EntityKeyUtil.findPrimaryKeyValue( key.getEntityKey() ); String edgeClassName =
+		  AssociationUtil.getMappedByFieldName( associationContext ); ORecordId outRid = (ORecordId)
+		  outEntityTuple.get( OrientDBConstant.SYSTEM_RID ); log.info( "removeAssociation: outRid:" + outRid +
+		  "; inClassName:" + inClassName + "; inBusinessPrimaryKeyName:" + inBusinessPrimaryKeyName +
+		  "; inBusinessPrimaryKeyValue:" + inBusinessPrimaryKeyValue + ";mappedBy:" + edgeClassName ); try { ORecordId
+		  inRid = EntityKeyUtil.findRid( provider.getConnection(), inClassName, inClassName, inBusinessPrimaryKeyValue
+		  ); AssociationUtil.removeAssociation( provider.getConnection(), edgeClassName, outRid, inRid ); } catch
+		  (SQLException sqle) { log.error( "Error!", sqle ); throw new RuntimeException(
+		  "Can not insert or update association", sqle ); }
+		 */
+	}
+
+	@Override
+	public boolean isStoredInEntityStructure(AssociationKeyMetadata associationKeyMetadata, AssociationTypeContext associationTypeContext) {
+		return true;
+	}
+
+	@Override
+	public Number nextValue(NextValueRequest request) {
+		log.info( "NextValueRequest:" + request + "; " );
+		// return ORecordId.EMPTY_RECORD_ID.getClusterPosition();
+		throw new UnsupportedOperationException( "nextValue Not supported yet." );
+	}
+
+	@Override
+	public boolean supportsSequences() {
+		return true;
+	}
+
+	@Override
+	public ClosableIterator<Tuple> executeBackendQuery(BackendQuery<String> backendQuery, QueryParameters queryParameters) {
+
+		Map<String, Object> parameters = getNamedParameterValuesConvertedByGridType( queryParameters );
+		String nativeQuery = buildNativeQuery( backendQuery, queryParameters );
+		try {
+			log.info( "executeBackendQuery.native query: " + nativeQuery );
+			PreparedStatement pstmt = provider.getConnection().prepareStatement( nativeQuery );
+			for ( Map.Entry<String, TypedGridValue> entry : queryParameters.getNamedParameters().entrySet() ) {
+				String key = entry.getKey();
+				TypedGridValue value = entry.getValue();
+				log.info( "key: " + key + "; type:" + value.getType().getName() + "; value:" + value.getValue() );
+				// @todo move to Map
+				if ( value.getType().getName().equals( "string" ) ) {
+					pstmt.setString( 1, (String) value.getValue() );
+				}
+				if ( value.getType().getName().equals( "long" ) ) {
+					pstmt.setLong( 1, (Long) value.getValue() );
+				}
```

Comment:
Does this only work for Long values and String?
We should try to add all the other values
(I think it would be easier to treat this as a seprate issue, but it's your call)